Added platform support documentation for Android (#811)
o3de/o3de.org: content/docs/user-guide/platforms/android/generating_android_project_windows.md

set O3DE_ANDROID_SIGNCONFIG_KEY_PASSWORD=o3de_android
set O3DE_ANDROID_SIGNCONFIG_KEY_SIZE=2048
set O3DE_ANDROID_SIGNCONFIG_VALIDITY_DAYS=10000
set O3DE_ANDROID_DN="cn=atom-sample-viewer, ou=o3de, o=LF, c=US"
```

{{< note >}}
While have signing keys is required to deploy to an android device, this step is optional. If you already have an existing JKS file generated earlier, and know the values for the store password, key password, alias, etc, then they can be used in place of these values. 
{{< /note >}}

**Android Project Generation Settings**

```
set O3DE_ANDROID_SDK_PATH=C:\Android\android-sdk
set O3DE_ANDROID_NDK_VERSION=22.*
set O3DE_ANDROID_SDK_API_LEVEL=29
set O3DE_ANDROID_ASSET_MODE=LOOSE
set O3DE_ANDROID_DEPLOY_TYPE=BOTH
```

{{< note >}}
The `O3DE_ANDROID_SDK_PATH` represents a specific location based on where you specify it in Android Studio or where you download and unpack the Android SDK Tools package
{{< /note >}}

<br/>

## Steps

{{< note >}}
Steps 1-3 is only needed if the windows build of O3DE has not been completed yet. These steps describe the minimal build necessary in order to process assets for Android and the game project.
{{< /note >}}

> **Step 1. Generate the cmake build directory for windows / vs2019 (for the android assets)**
> 
> ```
> cmake -B %O3DE_BUILD_ROOT%\windows -S %O3DE_ENGINE_PATH% -G"Visual Studio 16 2019" -DLY_PROJECTS=%O3DE_PROJECT_PATH% -DLY_UNITY_BUILD=ON
> ```
> 

Reviewer: - Change steps to numerical list. 
- Don't bold the steps.
- Indent the code to align it with the step. (You can see this when you render the page.)

```suggestion
1. Generate the Visual Studio project for the O3DE engine.

    ```
    cmake -B %O3DE_BUILD_ROOT%\windows -S %O3DE_ENGINE_PATH% -G"Visual Studio 16 2019" -DLY_PROJECTS=%O3DE_PROJECT_PATH% -DLY_UNITY_BUILD=ON
    ```
```

> **Step 2. Build the Asset Processor Batch project in profile mode to process the assets**
> 
> ```
> cmake --build %O3DE_BUILD_ROOT%\windows --config profile --target AssetProcessorBatch -- /m /nologo
> ```
> 
> {{< note >}}
> This step will only build 'AssetProcessorBatch' and the necessary dependent modules. You can alternatively build the entire suite of tools, libraries, etc for windows by replacing `AssetProcessorBatch` with `ALL_BUILD`, or alternatively build from Visual Studio directly.
> {{< /note >}}

Reviewer: - Removed redundant content in the note. We already tell the user they can build the full suite of tools in the beginning of this subsection. 

```suggestion
2. Build the **Asset Processor Batch** project in profile mode to process the assets.

    ```
    cmake --build %O3DE_BUILD_ROOT%\windows --config profile --target AssetProcessorBatch -- /m /nologo
    ```

    {{< note >}}
This step only builds Asset Processor Batch and the necessary dependent modules.
    {{< /note >}}
```

> 

Reviewer: - This section contains only the required steps needed to generate the Android project on Windows. 

```suggestion

## Generating Android projects on Windows  

The following instructions assume that you've completed the optional steps to set up environment variables and build O3DE and Asset Processor Batch.

```

> **Step 3. Run Asset Processor Batch for android and Atom Sample Viewer**
> 
> ```
> %O3DE_BUILD_ROOT%\windows\bin\profile\AssetProcessorBatch.exe  --platforms=android --project-path %O3DE_PROJECT_PATH%
> ```
> 

Reviewer: ```suggestion
1. Run Asset Processor Batch to process Android assets for the Atom Sample Viewer project.

    ```
    %O3DE_BUILD_ROOT%\windows\bin\profile\AssetProcessorBatch.exe --platforms=android --project-path %O3DE_PROJECT_PATH%
    ```
```

> **Step 4. Generate a key store to be able to self-sign an APK**
> 
> ```
> keytool -genkey -keystore %O3DE_ANDROID_SIGNCONFIG_FILE% -storepass %O3DE_ANDROID_SIGNCONFIG_STORE_PASSWORD% -alias %O3DE_ANDROID_SIGNCONFIG_KEY_ALIAS% -keypass %O3DE_ANDROID_SIGNCONFIG_KEY_PASSWORD% -keyalg RSA -keysize %O3DE_ANDROID_SIGNCONFIG_KEY_SIZE% -validity %O3DE_ANDROID_SIGNCONFIG_VALIDITY_DAYS% -dname %O3DE_ANDROID_DN%
> ```

Reviewer: - Marked this as optional and provided a clear alternative step. 
```suggestion
2. (Optional) Generate a KeyStore to allow APK signing. If you don't set up a signing key at this time, you can also set it up through Android Studio after generating the Android project.

    ```
    keytool -genkey -keystore %O3DE_ANDROID_SIGNCONFIG_FILE% -storepass %O3DE_ANDROID_SIGNCONFIG_STORE_PASSWORD% -alias %O3DE_ANDROID_SIGNCONFIG_KEY_ALIAS% -keypass %O3DE_ANDROID_SIGNCONFIG_KEY_PASSWORD% -keyalg RSA -keysize %O3DE_ANDROID_SIGNCONFIG_KEY_SIZE% -validity %O3DE_ANDROID_SIGNCONFIG_VALIDITY_DAYS% -dname %O3DE_ANDROID_DN%
    ```
```
```

> {{< note >}}
> Even though a signing key is required for the project in order to deploy, it does not necessary to provide this at project creation. This can also be achieved through Android Studio after project creation. If this is the case, then the generate android command in the following step can omit all of the --signconfig-* arguments, and the key store settings can be created through Android Studio after opening the generated android project.
> {{< /note >}}

Reviewer: - Removed this note and combined the content into the new step 2 and the new step 3. 
```suggestion
```

> **Step 5. Generate the android project**
> 
> ```
> %O3DE_ENGINE_PATH%\python\python.cmd %O3DE_ENGINE_PATH%\cmake\Tools\Platform\Android\generate_android_project.py --engine-root %O3DE_ENGINE_PATH% --project-path %O3DE_PROJECT_PATH% --build-dir %O3DE_BUILD_ROOT%\android --third-party-path %USERPROFILE%\.o3de\3rdParty --android-sdk-path %O3DE_ANDROID_SDK_PATH% --android-ndk-version %O3DE_ANDROID_NDK_VERSION%  --android-sdk-platform %O3DE_ANDROID_SDK_API_LEVEL% --include-apk-assets --asset-mode %O3DE_ANDROID_ASSET_MODE% --signconfig-store-file %O3DE_ANDROID_SIGNCONFIG_FILE% --signconfig-store-password %O3DE_ANDROID_SIGNCONFIG_STORE_PASSWORD% --signconfig-key-alias %O3DE_ANDROID_SIGNCONFIG_KEY_ALIAS% --signconfig-key-password %O3DE_ANDROID_SIGNCONFIG_KEY_PASSWORD% --overwrite-existing
> ```

Reviewer: I updated this step to make it more clear what the user's next action step be. Their next step depends on whether they generated a KeyStore file or not (see new step 2).

```suggestion
3. Use Python to run `generate_android_project.py` and generate the Android project for Atom Sample Viewer. The following command by itself assumes you skipped step 2 and that you plan to create KeyStore settings through Android Studio after generating the project. Otherwise, if you did generate a KeyStore in step 2, then include the `--signconfig-*` options listed below and specify their values.

    ```
    %O3DE_ENGINE_PATH%\python\python.cmd %O3DE_ENGINE_PATH%\cmake\Tools\Platform\Android\generate_android_project.py --engine-root %O3DE_ENGINE_PATH% --project-path %O3DE_PROJECT_PATH% --build-dir %O3DE_BUILD_ROOT%\android --third-party-path %USERPROFILE%\.o3de\3rdParty --android-sdk-path %O3DE_ANDROID_SDK_PATH% --android-ndk-version %O3DE_ANDROID_NDK_VERSION% --android-sdk-platform %O3DE_ANDROID_SDK_API_LEVEL% --include-apk-assets --asset-mode %O3DE_ANDROID_ASSET_MODE%
    ```
```

> {{< note >}}
> At this point, you can alternatively open the folder %O3DE_BUILD_ROOT%\android directly in Android Studio and build the APK from there instead of using command line in the next step.
> {{< /note >}}

Reviewer: - Moved and revised this note to new step 4 (old step 6, Build the Android project). 

```suggestion
```

> **Step 6. Build the android project**
> 
> ```
> cd %O3DE_BUILD_ROOT%\android
> 
> gradlew assembleProfile
> ```

Reviewer: ```suggestion
4. Build the Android project.

    ```
    cd %O3DE_BUILD_ROOT%\android

    gradlew assembleProfile
    ```
```

> 

Reviewer: - Make sure that note is indented so that it lines up underneath step 4. 
```suggestion

    {{< note >}}
Alternatively, instead of using command line, you can open the folder `%O3DE_BUILD_ROOT%\android` directly in Android Studio and build the APK from there.
    {{< /note >}}

```

> **Step 7. Deploy to the android device**
> 
> ```
> %O3DE_ENGINE_PATH%\python\python.cmd %O3DE_ENGINE_PATH%\cmake\Tools\Platform\Android\deploy_android.py --build-dir %O3DE_BUILD_ROOT%\android --configuration profile --clean -t %O3DE_ANDROID_DEPLOY_TYPE%
> ```
> {{< note >}}
> The deployment tool will rely on android's adb tool to perform the deployment. The target device must be placed in development mode, and connected to USB. Follow the [Android instructions](https://developer.android.com/studio/debug/dev-options) on how to do this. You may be ask to trust the host machine when deploying, and the deployment process will be blocked until you enable the device to trust the host machine.
> {{< /note >}}

Reviewer: ```suggestion
5. Deploy your project to an Android device.

    ```
    %O3DE_ENGINE_PATH%\python\python.cmd %O3DE_ENGINE_PATH%\cmake\Tools\Platform\Android\deploy_android.py --build-dir %O3DE_BUILD_ROOT%\android --configuration profile --clean -t %O3DE_ANDROID_DEPLOY_TYPE%
    ```
    {{< note >}}
The deployment tool relies on Android Debug Bridge (ADB), Android's command-line tool, to perform the deployment. You must set the target device to development mode and connect it to your machine using a USB. To set this up, follow the [Android instructions](https://developer.android.com/studio/debug/dev-options). Your target device may ask you to trust the host machine, which must enable to continue the deployment process.
    {{< /note >}}
```
